lib/python/rosie/ws_client_auth.py

```diff
@@ -22,22 +22,35 @@
 
 import ast
 from getpass import getpass
-try:
-    import gnomekeyring
-    import gtk
-except (ImportError, RuntimeError):
-    pass
 import os
 import re
+import shlex
+import sys
+from urlparse import urlparse
+
 import rose.config
 from rose.env import env_var_process
 from rose.popen import RosePopener
 from rose.reporter import Reporter
 from rose.resource import ResourceLocator
-import shlex
+
 import socket
-import sys
-from urlparse import urlparse
+try:
+    from gi import require_version, pygtkcompat
+    require_version('Gtk', '3.0')  # For GTK+ >=v3 use PyGObject; v2 use PyGTK
+    require_version('Secret', '1')
+    from gi.repository import Secret
+    gi_flag = True
```

Comment:
This is a constant so should really use uppercase.

Reply:
Good point! Will get it changed now.